homeassistant/components/binary_sensor/ads.py

```diff
@@ -0,0 +1,78 @@
+"""Support for ADS binary sensors."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import BinarySensorDevice, \
+    PLATFORM_SCHEMA, DEVICE_CLASSES_SCHEMA
+from homeassistant.components.ads import DATA_ADS, CONF_ADSVAR
+from homeassistant.const import CONF_NAME, CONF_DEVICE_CLASS
+import homeassistant.helpers.config_validation as cv
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['ads']
+DEFAULT_NAME = 'ADS binary sensor'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADSVAR): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Binary Sensor platform for ADS."""
+    ads_hub = hass.data.get(DATA_ADS)
+    if not ads_hub:
```

Comment:
This shouldn't be able to happen, since this platform depends on the ads component, which adds the `DATA_ADS` in `hass.data`.

Reply:
It is possible that the ADS component can not be created if a communication with the device can not be established. In this case I need to check for a valid component. Does Homeassistant do this prior to calling the `setup_platform` function? In this case I would agree to remove this check.

Reply:
I just found the answer in the documentation. So I will remove this check.